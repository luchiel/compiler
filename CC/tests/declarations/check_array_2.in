int main()
{
int a[6][7];
a[3];
}