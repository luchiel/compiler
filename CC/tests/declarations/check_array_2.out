type <int>
type <float>
type <void>
type <int*>
	pointer to <int>
type <void*>
	pointer to <void>
variable NULL of type <void*> initialized by
	{int 0}
		type <int>
function printf
	returns type <int>
	arguments
		<no args>
function main
	returns type <int>
	arguments
		<no args>
	{{}}
		type <0>
			array [
				{int 6}
					type <int>
			] of array [
				{int 7}
					type <int>
			] of <int>
		variable a of type <0>
	| 
	+-{;}
	  | 
	  +-{[]}
					array [
						{int 7}
							type <int>
					] of <int>
	    | 
	    +-{a}
					type <0>
						array [
							{int 6}
								type <int>
						] of array [
							{int 7}
								type <int>
						] of <int>
	    | 
	    +-{int 3}
					type <int>
